### PR TITLE
Foreign support

### PR DESCRIPTION
**Pull request by toastal**

### CODE REVIEW

**Comment by srid** [comment]
@toastal I pushed a fix about the Nix issue.

Here's what I see when running the new baz application:

```console
❯ nix --option system x86_64-darwin run .#baz
+ /nix/store/7m8as7wrzy9pwbcpgzhg21rlqqg1nmky-nodejs-18.12.1/bin/node /nix/store/z9jkj37ms37i7rr8426w4c00na9327d5-Main-bundle
/nix/store/z9jkj37ms37i7rr8426w4c00na9327d5-Main-bundle:781
  return (0, import_zalgo_js.zalgo)(str, { seed: "PureScript \u222A Nix" });
                                   ^

TypeError: (0 , import_zalgo_js.zalgo) is not a function
    at zalgoize (/nix/store/z9jkj37ms37i7rr8426w4c00na9327d5-Main-bundle:781:36)
    at Object.<anonymous> (/nix/store/z9jkj37ms37i7rr8426w4c00na9327d5-Main-bundle:1932:48)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47

Node.js v18.12.1
```

By the way, I made you a collaborator - so if you re-open this PR by using a branch pushed to this very repo, the CI (Garnix) will run.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+[*]
+end_of_line = lf
+insert_final_newline = true
+
+[*.js]
+indent_style = tab
+
+[*.{json,nix,purs,yaml,yml}]
+indent_style = space
+indent_size = 2
```
Comment (srid) [inline .editorconfig]: We should get rid of this file from this PR, and perhaps make it part of its own PR. That said I don't think I want `.editorconfig` on this repo yet ...

Comment (toastal) [reply .editorconfig]: I can drop this, but I wanted/needed it for my editor situation. My editor default do not match this repository.

**Review by srid — COMMENTED** [review]
@toastal The code changes look good (I also pushed a few changes you can check out if you're curious). 

`nix run .#qux` works. However, while `cd ./bar && purs-nix run` works (in dev shell), `cd ./qux && purs-nix run` does not. 

```
node:internal/errors:484
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'zalgo-js' imported from /Users/srid/code/purescript-multi/output/Main/foreign.js
    at new NodeError (node:internal/errors:393:5)
    at packageResolve (node:internal/modules/esm/resolve:860:9)
    at moduleResolve (node:internal/modules/esm/resolve:909:20)
    at defaultResolve (node:internal/modules/esm/resolve:1124:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:841:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.12.1
```

Did we hit a purs-nix bug? cc @ursi

**Comment by ursi** [comment]
@srid `cd qux && purs-nix compile` works for me

**Comment by ursi** [comment]
ah, you meant `purs-nix bundle`

**Comment by srid** [comment]
`purs-nix run`, basically. 

I haven't tried the bundle command, yet.

**Comment by srid** [comment]
I meant `purs-nix run`, but I made a typo (I've edited my comment now).

I haven't tried the bundle command, yet.

**Comment by toastal** [comment]
Funny, I renamed `baz` to `zalgo` for it to make more sense. `baz` comes after `bar` though, not `qux` if we're being pedantic.

**Comment by toastal** [comment]
Also, I'm not convinced "DRY"ing the code base in these examples is necessarily good. It adds a layer of indirection that makes it less clear, especially for Nix novices that just want something to copy so they can get back to PureScript (which should likely be the intended audience).

**Review by srid — APPROVED** [review]

**Comment by srid** [comment]
c82f78c006c899be243b995f66a6ffcba82ed412 fixed it!
